bioel/ontology.py

```diff
@@ -0,0 +1,212 @@
+from dataclasses import dataclass, field, asdict
+from typing import List, Optional, Union
+from tqdm import tqdm
+
+import obonet
+
+from .utils.obo_utils import _obo_extract_definition, _obo_extract_synonyms
+from .logger import setup_logger
+
+from .utils.umls_utils import UmlsMappings
+
+logger = setup_logger()
+
+
+@dataclass
+class BiomedicalEntity:
+    """
+    Class for keeping track of all relevant fields in an ontology
+    """
+    cui: str
+    name: str
+    types: List[str]
+    aliases: List[str] = field(default_factory=list)
+    definition: Optional[str] = None
+    equivalant_cuis: Optional[List[str]] = None
+    taxonomy: Optional[str] = None
+    metadata: Optional[dict] = None
+
+
+@dataclass
+class BiomedicalOntology:
+    name: str
+    types: List[str] = field(default_factory=list)
+    entities: List[BiomedicalEntity] = field(default_factory=list) # Dict mapping CUI: BiomedicalEntity
```

Comment:
Just a thought, wouldn't it be better to organize the Ontology as a map from cui -> entity data using a dict data structure? We would still have list-like access from entities.values() but we would also have O(1) access to any entity and it's attributes. 

Not sure if this is a need right now, but if ever in this package we need to query a particular entity in an ontology, the current implementation with O(N) time complexity is very in-efficient.

Reply:
That's a nice point. I will make the above changes to OBO and UMLS.

Reply:
@ChristopheYe  Can you follow the above implementation for Medic and Mesh.

Reply:
Done with this change. I am merging this PR into the main.